Running tests now requires compilemessages

I believe PR #2213 may have introduced the need for the French translations to exist. Running `make test` with a fresh clone and following the README instructions will result in the following error (or deleting the relevant `docs/locale/fr/*.mo` files):

```
❯ python manage.py test --keepdb djangoproject.tests.LocaleSmokeTests
Found 3 test(s).
Using existing test database for alias 'default'...
System check identified no issues (1 silenced).
.F.
======================================================================
FAIL: test_docs_locale (djangoproject.tests.LocaleSmokeTests.test_docs_locale)
docs/locale/ should contain translations for 'Using Django'
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/schillingt/Projects/djangoproject.com/djangoproject/tests.py", line 33, in test_docs_locale
    self.assertEqual(
    ~~~~~~~~~~~~~~~~^
        translated,
        ^^^^^^^^^^^
        "Utilisation de Django",
        ^^^^^^^^^^^^^^^^^^^^^^^^
        msg="docs/locale/ translation not loaded or incorrect",
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
AssertionError: 'Using Django' != 'Utilisation de Django'
- Using Django
+ Utilisation de Django
 : docs/locale/ translation not loaded or incorrect

----------------------------------------------------------------------
Ran 3 tests in 0.012s

FAILED (failures=1)
Preserving test database for alias 'default'...
```

This works in CI because `make ci` first runs `make compilemessages`. I think updating the README set up docs to also include `make compilemessages` would be sufficient.

--- a/README.rst
+++ b/README.rst
@@ -81,6 +81,13 @@ Install and run locally from a virtual environment
 
     python -m manage loaddata dev_sites
 
+#. Compile .po files to .mo files for translations::
+
+    make compilemessages
+
+    This only needs to happen once to support running the locale tests
+    locally.
+
 #. For docs (next step requires ``gettext``)::
 
     python -m manage loaddata doc_releases

